Sources/Overload/OvCore/include/OvCore/Scripting/IScriptContext.h

```diff
@@ -0,0 +1,22 @@
+/**
+* @project: Overload
+* @author: Overload Tech.
+* @licence: MIT
+*/
+
+#pragma once
+
+namespace OvCore::Scripting
+{
+	/**
+	* Interface for scripting backends to implement and store script contexts
+	*/
+	class IScriptContext
+	{
+	public:
+		/**
+		* Returns true if the context is valid
+		*/
+		virtual bool IsValid() const = 0;
```

Comment:
To be truly an interface I believe we still need to make the destructor virtual with an empty body, otherwise we're pretty much abstract.

```suggestion
	public:
	        virtual ~IScriptContext() {}
		/**
		* Returns true if the context is valid
		*/
		virtual bool IsValid() const = 0;
```

Reply:
Indeed, a virtual destructor is required to avoid undefined behavior; however, I would suggest using the = default keyword for clarity and efficiency:

```suggestion
	public:
		virtual ~IScriptContext() = default;
```
This approach aligns with our modern C++ philosophy by being both concise and expressive.